### PR TITLE
OS Diff PR Workflow Tutorial

### PR DESCRIPTION
**Pull request by leoebfolsom**
Tutorial for dbt users who are using os data-diff to improve their pr workflow.

### CODE REVIEW

File: docs/os_diff/how_to_use/how_to_use_with_toml.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ There are two main sections:
 
 By setting up a TOML file like this:
 
-```toml
+```python
```
Comment (erezsh) [inline docs/os_diff/how_to_use/how_to_use_with_toml.md:27]: You changed one instance, but they are more...

File: docs/os_diff/os_diff_pr_tutorial.md [hunk 2]
```diff
@@ -0,0 +1,283 @@
+# Running `data-diff` to improve your PR review workflow
```
Comment (erezsh) [inline docs/os_diff/os_diff_pr_tutorial.md:1]: Does everyone know what a PR is?

Maybe better to call it "commit" or "code change"? Or even "migration".

File: docs/os_diff/os_diff_pr_tutorial.md [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+
+Let's walk through the process of using Datafold's open source `data-diff` tool to make sure the code changes in your PR don't lead to unexpected data changes in prod.
+
+## How does `data-diff` work?
+
+`data-diff` compares _every data value_ that is generated by your PR branch code and compares those values with the data in production.
```
Comment (erezsh) [inline docs/os_diff/os_diff_pr_tutorial.md:7]: Only on specified columns

File: docs/os_diff/os_diff_pr_tutorial.md [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+## Enough preamble. Let's diff.
+
+<p align="center">
+  <a href=https://www.loom.com/share/682e4b7d74e84eb4824b983311f0a3b2 target="_blank">
+    <img alt="Intro to Diff" src="https://user-images.githubusercontent.com/1799931/196576582-d3535395-12ef-40fd-bbbb-e205ccae1159.png" width="50%" height="50%" />
+  </a>
+</p>
+
+### Install `data-diff`.
+We recommend installing `data-diff` in the same virtual environment that you use when running dbt CLI.
```
Comment (erezsh) [inline docs/os_diff/os_diff_pr_tutorial.md:43]: If this is DBT heavy, better write it in the title or foreword.

File: docs/os_diff/os_diff_pr_tutorial.md [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+```
+
+A few notes about this command:
+
+- Everything in carrots (`<>`) should be replaced by your values, _including the carrots_.
+  - In other words, your command should include no carrots! 🚫🥕
+- Each line beginning with a hyphen represents an _option_. In the command above, we've included the main options you will likely need:
+  - `-k` specifies the name of your primary key column.
+  - `-c` specifies the other columns you want to inspect for differences
+    - To simplify your output, we recommend including only the columns that have altered lines of code in your PR.
```
Comment (erezsh) [inline docs/os_diff/os_diff_pr_tutorial.md:77]: And performance!

File: docs/os_diff/os_diff_pr_tutorial.md [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+- 123 rows changed from having string values of "UNDER REVIEW" or "PENDING" in Prod to being `NULL` in the PR.
+  - These are the primary keys that are missing from the PR.
+
+By materializing the results of our `data-diff` in the warehouse, we gained insight into data changes on the level of individual values, as well as the ability to quickly summarize the Test Results Table into meaningful conclusions about the impact the PR code will have on the data.
+
+Once we've completed this analysis, we can make any changes needed to the code, run `data-diff` again, and finally, ask for a PR review.
+
+#### Here's the best part.
+
+When you ask for a PR review, you can say: "Hi, can you please review this PR? I've already run a `data-diff`, so you don't need to worry about any unexpected impacts on the data."
```
Comment (erezsh) [inline docs/os_diff/os_diff_pr_tutorial.md:259]: Good punchline!

**Comment by leoebfolsom** [comment]
Thanks for the feedback, @erezsh . You were too quick and I didn't have a chance to explain the context. 

This is going to be sent to a small number (2-3) of potential customers who are looking to use data-diff _in the context of their dbt PR workflows_. @williebsweet had the same feedback as you, that for a broader audience, we'll want to not go so heavy on the PR language.

I'm going to incorporate your feedback into the current draft which will be sent to these potential clients; but for now, I'm closing this PR. Once we have a chance to get their feedback, I'll open a new PR for storing this tutorial in the public documentation site.

I initially thought we'd be using this repo to host this tutorial (and we will eventually, but just not for sending it to these initial prospective customers). My apologies for the confusion.

**Comment by erezsh** [comment]
@leoebfolsom  Can you still fix the TOML highlighting?

**Comment by leoebfolsom** [comment]
@erezsh sure, in a separate PR

**Pull request closed without merging**